packages/browser/test/unit/profiling/hubextensions.test.ts

```diff
@@ -0,0 +1,91 @@
+import { TextDecoder, TextEncoder } from 'util';
+// @ts-ignore patch the encoder on the window, else importing JSDOM fails (deleted in afterAll)
+const patchedEncoder = (!global.window.TextEncoder && (global.window.TextEncoder = TextEncoder)) || true;
+// @ts-ignore patch the encoder on the window, else importing JSDOM fails (deleted in afterAll)
+const patchedDecoder = (!global.window.TextDecoder && (global.window.TextDecoder = TextDecoder)) || true;
+
+import { getCurrentHub } from '@sentry/core';
+import type { Transaction } from '@sentry/types';
+import { JSDOM } from 'jsdom';
+
+import { onProfilingStartRouteTransaction } from '../../../src';
+
+// @ts-ignore store a reference so we can reset it later
+const globalDocument = global.document;
+// @ts-ignore store a reference so we can reset it later
+const globalWindow = global.window;
+// @ts-ignore store a reference so we can reset it later
+const globalLocation = global.location;
+
+describe('BrowserProfilingIntegration', () => {
+  beforeEach(() => {
+    const dom = new JSDOM();
+    // @ts-ignore need to override global document
+    global.document = dom.window.document;
+    // @ts-ignore need to override global document
+    global.window = dom.window;
+    // @ts-ignore need to override global document
+    global.location = dom.window.location;
+
+    const hub = getCurrentHub();
+    const client: any = {
+      getDsn() {
+        return {};
+      },
+      getTransport() {
+        return {
+          send() {},
+        };
+      },
+      getOptions() {
+        return {
+            profilesSampleRate: 1,
+        }
+      }
+    };
+
+    hub.bindClient(client);
+  });
+
+  // Reset back to previous values
+  afterEach(() => {
+    // @ts-ignore need to override global document
+    global.document = globalDocument;
+    // @ts-ignore need to override global document
+    global.window = globalWindow;
+    // @ts-ignore need to override global document
+    global.location = globalLocation;
+  });
+  afterAll(() => {
+    // @ts-ignore patch the encoder on the window, else importing JSDOM fails
+    patchedEncoder && delete global.window.TextEncoder;
+    // @ts-ignore patch the encoder on the window, else importing JSDOM fails
+    patchedDecoder && delete global.window.TextDecoder;
+  });
+
+  it('does not throw if Profiler is not available', () => {
+    // @ts-ignore force api to be undefined
+    global.window.Profiler = undefined;
```

Comment:
Do we need to reset this like we do with document, window and location?

Reply:
@lforst I think this is already done in beforeEach where we override the global.window object with the new one from the fresh jsdom instance

Reply:
Ah right. Didn't see that Profiler was on the window object. Sorry about that!

Reply:
All good :D always better to ask then to miss it :D